### ISSUE TITLE
Server port is not always reliable under django and nginx

### ISSUE THREAD

**Issue by patrick91** [onelogin/python3-saml#10]
Hi everyone. I've been having an issue with the redirect url, my setup is more or less this one (I'm using docker, but this is not related to it):

1) docker container with the backend running under uwsgi
2) docker container with nginx acting as a server proxy to the first container

The nginx container uses the default port 80 meanwhile the uwsgi is using the port 8000.

Here is the function that creates the redirect url: https://github.com/onelogin/python3-saml/blob/master/src/onelogin/saml2/utils.py#L226

I can see that it is using the `server_port` value, which can be wrong sometimes, like in this case.

Django 1.9 has a `get_port` function which takes care of the specific settings https://github.com/django/django/blob/6f1318734f0f3b6e62b782b0251a4e676e542e0b/django/http/request.py#L112

I'm not sure if this can be fixed via a nginx configuration, I've tried a couple of times with no luck.

I'm happy to submit a PR but I need some guidance on the project since I'm fairly new to it :)

**Comment by pitbulk** [comment]
We can't use a django method on the toolkit since it will be used by others frameworks.

You can see that we are using the server_port value, but this came from the request_data var.
This request_data is set on the [prepare_django_request method](https://github.com/onelogin/python3-saml/blob/master/demo-django/demo/views.py#L18) so is there were you may set the right value.

Maybe replace
```
'server_port': request.META['SERVER_PORT'], 
```
by
```
'server_port': , request.get_port(),
```

**Comment by patrick91** [comment]
Ohh, I didn't see that! Thanks!

Do you want me to submit a PR to change the sample to use that method instead of using the request.META dict?

**Comment by pitbulk** [comment]
If you want to contribute that, I'm ok. We can replace all request.META['xxx'] by its django specific method, (then you can remove the  comment:  
# If server is behind proxys or balancers use the HTTP_X_FORWARDED fields

**Comment by patrick91** [comment]
@pitbulk maybe it's better not doing this since the get_port method is available only from Django 1.9
